lib/Sema/ResilienceDiagnostics.cpp

```diff
@@ -0,0 +1,103 @@
+//===--- ResilienceDiagnostics.cpp - Resilience Inlineability Diagnostics -===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements diagnostics for @inlineable.
+//
+//===----------------------------------------------------------------------===//
+
+#include "TypeChecker.h"
+#include "swift/AST/Attr.h"
+#include "swift/AST/Decl.h"
+#include "swift/AST/Initializer.h"
+#include "swift/AST/DeclContext.h"
+using namespace swift;
+
+enum FragileFunctionKind : unsigned {
+  Transparent,
+  InlineAlways,
+  Inlineable,
+  DefaultArgument
+};
+
+/// If the body is able to be inlined into functions in other resilience
+/// domains, this ensures that only sufficiently-conservative access patterns
```

Comment:
I just removed the comment since it got copy-pasted from getResilienceExpansion() on accident.